Rest/Field/TypeValue.php

```diff
@@ -0,0 +1,181 @@
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class TypeValue
+{
+    /** @var \Symfony\Component\HttpFoundation\RequestStack */
+    private $request;
+
+    /** @var \eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\ConfigResolver */
+    private $configResolver;
+
+    /** @var \eZ\Publish\Core\FieldType\XmlText\Converter\Html5 */
+    private $html5Converter;
+
+    /** @var \eZ\Bundle\EzPublishCoreBundle\Imagine\AliasGenerator */
+    protected $imageVariationService;
+
+    /** @var array */
+    public $fieldDefIdentifiers;
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\RequestStack $request
+     * @param \eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Configuration\ConfigResolver $configResolver
+     * @param \eZ\Publish\Core\FieldType\XmlText\Converter\Html5 $html5Converter
+     * @param \eZ\Bundle\EzPublishCoreBundle\Imagine\AliasGenerator $imageVariationService
+     */
+    public function __construct(
+        RequestStack $request,
+        ConfigResolver $configResolver,
+        Html5 $html5Converter,
+        ImageVariationService $imageVariationService
+    ) {
+        $this->request = $request;
+        $this->configResolver = $configResolver;
+        $this->html5Converter = $html5Converter;
+        $this->imageVariationService = $imageVariationService;
+    }
+
+    /**
+     * Prepares an array with field type identifiers.
+     * 
+     * @param \eZ\Publish\API\Repository\Values\ContentType\ContentType $contentType
+     */
+    public function setFieldDefs(ContentType $contentType)
+    {
+        foreach ($contentType->fieldDefinitions as $fieldDef) {
+            $this->fieldDefIdentifiers[$fieldDef->identifier] = $fieldDef->fieldTypeIdentifier;
+        }
+    }
+
+    /**
+     * Returns field definition.
+     *
+     * @param string $identifier
+     *
+     * @return mixed
+     */
+    public function getFieldDef($identifier)
+    {
+        return $this->fieldDefIdentifiers[$identifier];
+    }
+
+    /**
+     * Returns field definitions.
+     *
+     * @return mixed
+     */
+    public function getFieldDefs()
+    {
+        return $this->fieldDefIdentifiers;
+    }
+
+    /**
+     * Returns parsed field value.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Field $field
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
+     * @param string $language
+     * @param string $imageFieldIdentifier
+     *
+     * @return mixed
+     */
+    public function getFieldValue(Field $field, Content $content, $language, $imageFieldIdentifier)
+    {
+        $fieldType = $this->fieldDefIdentifiers[$field->fieldDefIdentifier];
+
+        return $this->$fieldType($field, $content, $language, $imageFieldIdentifier);
+    }
+
+    /**
+     * Default field value parsing.
+     *
+     * @param string $fieldName
+     * @param mixed $args
+     *
+     * @return string
+     */
+    public function __call($fieldName, $args)
```

Comment:
What about splitting service to smaller parts? From what I can understand, everything above could be placed into different service. What do you think?